src/debugger/server/Debugger.js

```diff
@@ -299,18 +339,119 @@ export class DebugServer {
     return false;
   }
 
-  /*
-    Displays PP error message, then waits for user to run the program to
-    continue (similar to a breakpoint).
-  */
+  /**
+   * This function is only used if the original source locations in the
+   * sourcemaps are relative. This will discover the correct prefixes
+   * to use when converting the relative path to absolute paths.
+   */
+  _findSourcemapPrefixes(sourceMaps: Array<SourceFile> | void) {
+    // If sourcemaps don't exist, set prefixes to empty string and break.
+    if (sourceMaps) {
+      for (let map of sourceMaps) {
+        if (map.sourceMapContents === undefined || map.sourceMapContents === "") {
+          this._sourcemapCommonPrefix = "";
+          this._sourcemapMapDifference = "";
+          return;
+        }
+      }
+    } else {
+      this._sourcemapCommonPrefix = "";
+      this._sourcemapMapDifference = "";
+      return;
+    }
+
+    // Extract common prefix and map difference
+    let originalSourcePaths = [];
+    let mapPaths = [];
+    for (let map of sourceMaps) {
+      invariant(map.sourceMapContents); // Checked above.
+      let parsed = JSON.parse(map.sourceMapContents);
+      // Two formats for sourcemaps exist.
+      if ("sections" in parsed) {
+        for (let section of parsed.sections) {
+          for (let source of section.map.sources) {
+            originalSourcePaths.push(getAbsoluteSourcePath(map.filePath, source));
+          }
+        }
+      } else {
+        for (let source of parsed.sources) {
+          originalSourcePaths.push(getAbsoluteSourcePath(map.filePath, source));
+        }
+      }
+      mapPaths.push(stripEmptyStringBookends(map.filePath.split("/")));
+    }
+
+    let originalSourceCommonPrefix = findCommonPrefix(originalSourcePaths);
+    let originalSourceCPElements = stripEmptyStringBookends(originalSourceCommonPrefix.split("/"));
+    let mapCommonPrefix = findCommonPrefix(mapPaths);
+    let mapCPElements = stripEmptyStringBookends(mapCommonPrefix.split("/"));
+
+    this._sourcemapCommonPrefix = findCommonPrefix([originalSourceCPElements, mapCPElements]);
+    this._sourcemapMapDifference = findMapDifference(this._sourcemapCommonPrefix, mapCommonPrefix);
+  }
+
+  _relativeToAbsolute(path: string): string {
+    let absolute;
+    if (this._useRootPrefix) {
```

Comment:
Why do you need _useRootPrefix flag? Can't you use "_sourcemapDirectoryRoot === undefined" to know this fact?